django/thunderstore/frontend/api/experimental/views/community_package_list.py

```diff
@@ -225,10 +226,18 @@ def paginate(
             cache_vary=self.get_full_cache_vary(params),
             cache_bust_condition=CacheBustCondition.any_package_updated,
         )
-        return paginator.get_page(params["page"]).object_list
+
+        # PageNotAnInteger error won't be raised here since deserializer
+        # has already checked that the page parameter is an integer.
+        try:
+            page = paginator.page(params["page"])
```

Comment:
I've not realized to ask before, but does this not crash if there's no page parameter (which should probably either return the first page or redirect to the URL of the first page)?

Reply:
[The deserializer](https://github.com/thunderstore-io/Thunderstore/blob/master/django/thunderstore/frontend/api/experimental/serializers/views.py#L188) defines 1 as a default value for page.